### ISSUE TITLE
What's the proper setting of P and K

### ISSUE THREAD

**Issue by ofishhaha** [VisualComputingInstitute/triplet-reid#19]
The expeirment is conducted based on the default setting of parameters.
Because of the limitation of memory, I can not test on P=32, and K=4.
I did the experiment based on P=25, K=4.
But looks like the loss can not decrease and the final result of mAP is less than 10%.

When test on K=2 , it works well. And result of mAP is about 50%.

Is it because some mistake I make? Since K = 4 is also the setting in your paper, do you have any idea why this happens?

Best wishes.

**Comment by Pandoro** [comment]
What data are you training on? Throughout most of our experiments we used
K=4 and never had problems with that value.

**Comment by ofishhaha** [comment]
Market 1501 is used for training

**Comment by lucasb-eyer** [comment]
To add to that, we have used varying values for P. For the paper experiments, P=18, but we have also used P=32 and various values in-between using this code-base and never had such problems. This is all on Market1501, so it sounds like something else is wrong.

**Comment by ofishhaha** [comment]
I found that I forgot to load the pretrained parameters of the model. Now I got proper results.
Thank you all

**Comment by lucasb-eyer** [comment]
Thanks for letting us know what the mistake was, and glad you found it out :)

**Comment by liangbh6** [comment]
@Pandoro @lucasb-eyer Hi, I doubt that Whether P=18, K=4 is the best choice for TriNet? How to set them if I only can train on smaller batch size like P*K<40?

**Comment by lucasb-eyer** [comment]
Hi @liangbh6, in our paper we did not verify that it is the best choice or not, we simply chose whatever "seems reasonable and fits GPU memory". I plan to do a study over P/K choices for my thesis soon.

For your case, P*K<40, I unfortunately cannot say more than "just try". Maybe P=13, K=3? or P=10, K=4? I honestly don't know.

**Comment by liangbh6** [comment]
Well, then I should set another validation set to choose proper P and K?

**Comment by lucasb-eyer** [comment]
Yup, that's exactly what we did in the paper (using MARS) to find out which loss to use. And once you found a good one, use the full train-set with that setting.

If you want to "just get it to work" (as opposed to "get best test-performance")  you could also just look at the train-batch-topk during training on the train-set to see whether it's converging or not. But note that this is not a good indicator of generalization.

**Comment by liangbh6** [comment]
Good idea. Thanks a lot.
